Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -578,17 +612,18 @@ public class PBXProjGenerator {
             }
         }
 
-        let groupPath: String = depth == 0 ? path.byRemovingBase(path: spec.basePath).string : path.lastComponent
         let group: PBXGroup
-        if let cachedGroup = groupsByPath[path] {
-            group = cachedGroup
+        if spec.options.mirrorsFileSystem {
```

Comment:
Once thing that would need to be checked for is the case where you define a source outside of the base path eg. `sources: ../directory`. Currently this would probably add every directory until the filesystem root

Reply:
Good catch